src/main/webapp/profLoader.js

```diff
@@ -0,0 +1,128 @@
+/* using IIFE https://flaviocopes.com/javascript-iife/ */
+(() => {
+  google.charts.load("current", { packages: ["corechart"] });
+  google.charts.load("current", { packages: ["bar"] });
+
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const profKey = urlParams.get("prof-key");
+  const profName = urlParams.get("prof-name");
+
+  function populateData() {
+    fetch(`/prof-data?prof-key=${profKey}`)
+      .then((response) => response.json())
+      .then((profInfo) => {
+        document.getElementById("prof-name").innerHTML = profName;
+        google.charts.setOnLoadCallback(() => {
+          makeDiffComparisonChart(profInfo);
+          makePerceptionComparisonChart(profInfo);
+        });
+      });
+  }
+
+  document.addEventListener("DOMContentLoaded", function () {
+    populateData();
+  });
+
+  function makeDiffComparisonChart(profData) {
+    const average = (list) =>
+      list.reduce((prev, curr) => prev + curr, 0) / list.length;
+
+    const courseList = [];
+    const difficultyAvgList = [];
+
+    for (let dataHolder of profData) {
+      courseList.push(dataHolder.term);
+      difficultyAvgList.push(
+        average(dummyGradeData.concat(dataHolder.difficultyList)) + 3
```

Comment:
What is this 3 for? Might be good to have a comment

Reply:
will add, its just to make the dummy data look different from the other data